src/AggregateFunctions/Streaming/AggregateFunctionGroupUniqArrayRetract.h

```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include <AggregateFunctions/IAggregateFunction.h>
+#include <AggregateFunctions/KeyHolderHelpers.h>
+#include <AggregateFunctions/Streaming/CountedValueMap.h>
+#include <Columns/ColumnArray.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Common/HashTable/HashTableKeyHolder.h>
+#include <Common/assert_cast.h>
+
+
+namespace DB
+{
+struct Settings;
+
+namespace Streaming
+{
+template <typename T>
+struct AggregateFunctionGroupUniqArrayRetractData
+{
+    using Map = CountedValueMap<T, false>;
```

Comment:
The unique or distinct don't require ordered data. we can use `unordered CountedMap` . for example:
```c++
template <typename T, bool maximum, typename KeyCompare = void, bool ordered = true>
class CountedValueMap
{
    using BTreeMap = absl::btree_map<T, uint32_t, Compare>;
    using STDMap = std::map<T, uint32_t, Compare>;
    using STDUnorderedMap = std::unotdered_map<T, uint32_t, Compare>, 
    using Container = std::conditional_t<ordered, std::conditional_t<std::is_nothrow_copy_constructible<Compare>::value, BTreeMap, STDMap>, STDUnorderedMap>;
    ...
    
    Container map;
}
```

Reply:
Use another implementation `CountedHashMap`

Reply:
i will add it in this issue https://github.com/timeplus-io/proton/issues/486